src/telemetry/TelemetryDialog.cpp

```diff
@@ -0,0 +1,167 @@
+
+ @file TelemetryDialog.cpp
+ @brief Define a dialog for requesting user permissions for sending telentry events.
+
+ Dmitry Vedenko
+ **********************************************************************/
+
+#include "TelemetryDialog.h"
+
+#include <wx/sstream.h>
+#include <wx/txtstrm.h>
+#include <wx/fs_mem.h>
+#include <wx/button.h>
+
+#include "ShuttleGui.h"
+#include "widgets/HelpSystem.h"
+
+#include "lib-string-utils/CodeConversions.h"
+
+#include "../../images/Telemetry/TelemetryEmojii.h"
+
+static const auto title =
+/* i18n-hint: Title of the telemetry permission dialog */
+XO ("Help us improve Audacity");
+
+static const auto firstParagraph =
+/* i18-hint: The first paragraph of telemetry permission dialog */
+XO (
+"We would like to collect anonymous usage data to \
+help us prioritize improvements and making Audacity \
+better in the future. This includes things like \
+frequency of use, preferred file formats, crash logs etc."
+);
+
+static const auto secondParagraph =
+/* i18-hint: The second paragraph of telemetry permission dialog */
+XO (
+"We <u>do not</u> collect any personal data \
+or sensitive information such as location or \
+file names or any content of your audio. \
+<a href=\"https://www.audacityteam.org/contact/privacy-policy/\">Privacy policy</a>"
+);
+
+static const auto thirdParagraph =
+/* i18-hint: Hint to the user about how to change the setting it telemetry permission dialog */
+XO (
+    "You can change this at any time in <b>Preferences > Analytics</b>."
+);
+
+
+
+static const auto acceptButtonTitle =
+/* i18-hint: Accept button it telemetry permission dialog */
+XO ("Send anonymous analytics data");
+
+static const auto rejectButtonTitle =
+/* i18-hint: Reject button it telemetry permission dialog */
+XO ("Don't send");
+
+
+BEGIN_EVENT_TABLE (TelemetryDialog, wxDialogWrapper)
+    EVT_BUTTON (wxID_YES, TelemetryDialog::OnAccept)
+    EVT_BUTTON (wxID_NO, TelemetryDialog::OnDecline)
+END_EVENT_TABLE ()
+
+IMPLEMENT_CLASS (TelemetryDialog, wxDialogWrapper)
+
+TelemetryDialog::TelemetryDialog (wxWindow* parent)
+/* i18n-hint: information about the anonymous data collection */
+    : wxDialogWrapper (parent, -1, XO ("Analytics & crash logging"),
+        wxDefaultPosition, wxDefaultSize,
+        wxCAPTION)
+{
+    ShuttleGui S (this, eIsCreating);
+
+    wxFileSystem::AddHandler (new wxMemoryFSHandler);
+
+    wxMemoryFSHandler::AddFile ("CheckMarkEmojii.png", bin2c_CheckMarkEmojii_png, sizeof (bin2c_CheckMarkEmojii_png));
+    wxMemoryFSHandler::AddFile ("HeadphonesEmojii.png", bin2c_HeadphonesEmojii_png, sizeof (bin2c_HeadphonesEmojii_png));
+
+    S.SetBorder (5);
+    S.StartVerticalLay (wxEXPAND, 1);
+    {
+        S.AddWindow (CreateHTMLWindow (S.GetParent ()));
+
+        S.StartHorizontalLay (wxEXPAND, 0);
+        {
+            S.Prop (1).AddSpace (1, 0, 1);
+
+            S.Id (wxID_NO).AddButton (rejectButtonTitle);
+            S.Id (wxID_YES).AddButton (acceptButtonTitle)->SetDefault ();
```

Comment:
> Consent should be given by a clear affirmative act establishing a freely given, specific, informed and unambiguous indication of the data subject's agreement to the processing of personal data relating to him or her, such as by a written statement, including by electronic means, or an oral statement. This could include ticking a box when visiting an internet website, choosing technical settings for information society services or another statement or conduct which clearly indicates in this context the data subject's acceptance of the proposed processing of his or her personal data.
> **Silence, pre-ticked boxes or inactivity should not therefore constitute consent.**

[GDPR regulation](https://eur-lex.europa.eu/legal-content/EN/TXT/HTML/?uri=CELEX:32016R0679&from=EN)